### PR TITLE
Allow conanfile path in path argument for conan build, source, test, install, info

### PR DESCRIPTION
**Pull request by danimtb**
Issue #2143 for **source** command.

**TODO:** 

- [ ] `conan imports`
- [ ] `conan get`

### CODE REVIEW

File: conans/test/integration/symlinks_test.py [hunk 1]
```diff
@@ -66,13 +66,13 @@ def basic_test(self):
         client.save({"conanfile.py": conanfile,
                      "conanfile.txt": test_conanfile})
         client.run("export lasote/stable")
-        client.run("install --build -f=conanfile.txt")
+        client.run("install conanfile.py --build")
```
Comment (memsharded) [inline conans/test/integration/symlinks_test.py]: The original test was using conanfile.txt, not conanfile.py

File: conans/test/command/source_test.py [hunk 2]
```diff
@@ -165,7 +181,7 @@ def source(self):
 
         # Fix the error and repeat
         client.save({CONANFILE: conanfile.replace("err", "")})
-        client.run("source .")
+        client.run("source conanfile.py")
```
Comment (memsharded) [inline conans/test/command/source_test.py]: This test should change. ``conan source .`` is a valid command.

File: conans/test/command/build_test.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def build(self):
 """
         client.save({CONANFILE: conanfile_user_info}, clean_first=True)
         client.run("install --build=missing")
-        client.run("build .")
+        client.run("build ./conanfile.py")
```
Comment (memsharded) [inline conans/test/command/build_test.py]: Some some with ``build .``

Comment (danimtb) [reply conans/test/command/build_test.py]: Yes, I left some tests with just `build .` Check line 110 for example

**Pull request closed without merging**